boskos/cmd/cli/cli.go

```diff
@@ -240,6 +247,76 @@ Examples:
 	}
 	root.AddCommand(metrics)
 
+	heartbeat := &cobra.Command{
+		Use:   "heartbeat",
+		Short: "Send a heartbeat for a resource reservation",
+		Long: `Send a heartbeat for a resource reservation
+
+When the Boskos reaper is deployed, resource lessees must send a
+heartbeat for every lease they hold or the leases will be revoked.
+This command will send a heartbeat at the provided period and is
+blocking.
+
+Examples:
+
+  # Acquire one clean "my-thing" and mark it dirty when leasing
+  $ resource="$( boskosctl acquire --type my-thing --state clean --target-state dirty )"
+  # Send periodic heartbeat for the lease
+  $ boskosctl heartbeat --resource "${resource}" --period 30s &`,
+		Run: func(cmd *cobra.Command, args []string) {
+			options.initializeClient()
+			var resource common.Resource
+			if err := json.Unmarshal([]byte(options.heartbeat.resourceJSON), &resource); err != nil {
+				fmt.Fprintf(cmd.ErrOrStderr(), "failed to parse resource: %v\n", err)
+				exit(1)
+				return
+			}
+
+			ctx, cancel := context.WithTimeout(context.Background(), options.heartbeat.timeout)
+			sig := make(chan os.Signal, 1)
+			signal.Notify(sig, os.Interrupt)
+			go func() {
+				<-sig
+			}()
+
+			tick := time.Tick(options.heartbeat.period)
+		out:
```

Comment:
hummmmmmmmmm won't `defer cancel()` work? this seems not a great pattern?

Reply:
we still need to break out of the `for{}`, could `defer cancel()` but will need the label (or an anonymous func) anyway